x-pack/plugins/apm/public/components/app/onboarding/index.tsx

```diff
@@ -69,7 +71,24 @@ export function Onboarding() {
     } finally {
       setApiKeyLoading(false);
     }
-  }, []);
+  };
+
+  const checkAgentStatus = async () => {
+    try {
+      setAgentStatusLoading(true);
+      const agentStatusCheck = await callApmApi(
+        'GET /internal/apm/agent_status',
+        {
+          signal: null,
+        }
+      );
+      setAgentStatus(agentStatusCheck.status);
+    } catch (error) {
+      setAgentStatus(false);
```

Comment:
Do you need this if you set it in `finally`?

Reply:
Yup, reason being we have 3 situations.
1. Initially when agent status is not present then the value is undefined. No Callouts displayed
2. In case of Error (this one) and No Data present, both should behave as `false`. Callout displayed stating - No data available
3. Only in case of success, `true`, Callout displaying Data available